icu4c/source/i18n/number_modifiers.cpp

```diff
@@ -60,6 +60,41 @@ Modifier::Parameters::Parameters(
     const ModifierStore* _obj, Signum _signum, StandardPlural::Form _plural)
         : obj(_obj), signum(_signum), plural(_plural) {}
 
+bool Modifier::semanticallyEquivalent(const Modifier& other) const {
+    Parameters params0;
+    Parameters params1;
+    getParameters(params0);
+    other.getParameters(params1);
+    if (params0.obj == nullptr && params1.obj == nullptr) {
+        return strictEquals(other);
+    } else if (params0.obj == nullptr || params1.obj == nullptr) {
+        return false;
+    }
+    for (size_t i=0; i<SIGNUM_COUNT; i++) {
+        auto signum = static_cast<Signum>(i);
+        for (size_t j=0; j<StandardPlural::COUNT; j++) {
+            auto plural = static_cast<StandardPlural::Form>(j);
+            auto mod1 = params0.obj->getModifier(signum, plural);
+            auto mod2 = params1.obj->getModifier(signum, plural);
```

Comment:
similarly, consider using prefixes or suffixes like "this" and "other" rather than 1 and 2. especially when mod1 is from params0 and mod2 is from params1.....